### ISSUE TITLE
Wrong layout when rotating screen

### ISSUE THREAD

**Issue by flowbe** [scenee/FloatingPanel#515]
### Description

We recently transitioned one of our floating panels from a `UITableView` to a `UICollectionView` using compositional layouts. After transitioning, we noticed that when rotating the screen (portrait > landscape > portrait), our layout would keep the landscape size. We didn't have this issue before.

| Before | After |
|--------|-------|
| ![Before](https://user-images.githubusercontent.com/8288625/142632309-f68d0475-b070-43fe-89ba-a7da8f8fbf08.png) | ![After](https://user-images.githubusercontent.com/8288625/142632359-bb0e340f-8636-4f0d-8daf-b9e6e70eb8f8.png) |

We tried updating both the collection view layout and the panel layout in `viewWillTransition` but nothing works. Could this be a bug related to collection views?

### Expected behavior

Update the layout width accordingly to the orientation.

### Actual behavior

When switching from landscape to portrait, the landscape width is kept.

### Steps to reproduce

**Code example that reproduces the issue**

Our collection view layout looks like this:
```swift
let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .fractionalHeight(1))
let item = NSCollectionLayoutItem(layoutSize: itemSize)
item.contentInsets = NSDirectionalEdgeInsets(top: 5, leading: 5, bottom: 5, trailing: 5)
let groupSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .estimated(100))
let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, subitem: item, count: 2)
group.contentInsets = NSDirectionalEdgeInsets(top: 0, leading: 15, bottom: 0, trailing: 15)
let section = NSCollectionLayoutSection(group: group)
let layout = UICollectionViewCompositionalLayout(section: section)
```

The complete code is available [here](https://github.com/Infomaniak/ios-kDrive/pull/409/files).

**How do you display panel(s)?**

* ~~Add as child view controllers~~
* Present modally

**How many panels do you displays?**

* 1 
* ~~2+~~

### Environment

**Library version**

2.5.0

**Installation method**

* ~~CocoaPods~~
* ~~Carthage~~
* Swift Package Manager
 
**iOS version(s)**

15.0

**Xcode version**

13.1

**Comment by flowbe** [comment]
After investigating, I noticed that the constraints on the content view are given a lower priority (`$0.priority = .required - 1`) which leads into the issue. When commenting the line, everything works fine. Why are you doing this?

**Comment by scenee** [comment]
@flowbe Could you please let me know whether your panel's`FloatingPanelController.contentMode` is `.static` or `. fitToBounds`?  The reason why the priority was set to `.required - 1` is #359, which fixed #294. However #294 was an issue on `. fitToBounds` content mode. So if your panel’s content mode is `.static`, I’d like to the priory will be set to `.required` on `.static` content mode.

**Comment by flowbe** [comment]
Hi @scenee, we didn't change the `contentMode` variable so it is set to the default value, which is apparently `.static`

**Comment by scenee** [comment]
@flowbe Thanks for your information. I will change the priority on `.static` content mode.

**Comment by flowbe** [comment]
Hi @scenee, I saw that you made a PR for this issue, thanks a lot! Can you create a new release so I can update my project?
Thanks a lot

**Comment by scenee** [comment]
I've released v2.5.1 now 👍 Thanks for your support and patient.
